### PR TITLE
Optionally catch exceptions in transformer

### PR DESCRIPTION
**Pull request by fabianvf**
@chrisseto @erinspace Ready for review

### CODE REVIEW

File: scrapi/base/transformer.py [hunk 1]
```diff
@@ -46,7 +56,7 @@ def _transform_value(self, value, doc):
 
     def _transform_list(self, l, doc):
         return [
-            self._transform_value(item, doc) for item in l
+            self._maybe_transform_value(item, doc) for item in l
```
Comment (fabianvf) [inline scrapi/base/transformer.py]: ```fail``` is lost here.

**Comment by erinspace** [comment]
aside from merge conflicts, looks good! :+1:

File: scrapi/base/transformer.py [hunk 2]
```diff
@@ -21,20 +21,30 @@ def _transform_string(self, string, doc):
...
 
-    def _transform_value(self, value, doc):
+    def _maybe_transform_value(self, value, doc, fail=False):
+        try:
+            return self._transform_value(value, doc, fail=fail)
+        except Exception as e:
+            if fail:
+                raise
+            logger.error('{} failed to transform'.format(value))
+            logger.exception(e)
```
Comment (erinspace) [inline scrapi/base/transformer.py]: Are both of these needed?

File: scrapi/base/transformer.py [hunk 3]
```diff
@@ -21,20 +21,30 @@ def _transform_string(self, string, doc):
     def schema(self):
         raise NotImplementedError
 
-    def transform(self, doc):
-        return self._transform(self.schema, doc)
+    def transform(self, doc, fail=False):
+        return self._transform(self.schema, doc, fail=fail)
 
-    def _transform(self, schema, doc):
+    def _transform(self, schema, doc, fail=False):
```
Comment (erinspace) [inline scrapi/base/transformer.py]: Maybe change this name to _transform_dict() ?

**Comment by erinspace** [comment]
Awesome.

**Comment by fabianvf** [comment]
@erinspace @chrisseto, someone want to do pass 2 and/or merge?

File: scrapi/base/__init__.py [hunk 4]
```diff
@@ -78,7 +78,7 @@ class JSONHarvester(BaseHarvester, JSONTransformer):
     file_format = 'json'
 
     def normalize(self, raw_doc):
-        transformed = self.transform(json.loads(raw_doc['doc']))
+        transformed = self.transform(json.loads(raw_doc['doc']), fail=settings.RAISE_IN_TRANSFORMER)
```
Comment (chrisseto) [inline scrapi/base/__init__.py]: Will we ever want to change this after the application has started?

File: scrapi/settings/defaults.py [hunk 5]
```diff
@@ -18,44 +20,3 @@
 }
 
 DAYS_BACK = 2
-
-SCRAPI_URL = 'http://173.255.232.219'
```
Comment (chrisseto) [inline scrapi/settings/defaults.py]: Like

**Pull request merged**